### PR TITLE
[Test] Validate RESTful operations on OSD endpoint

### PR DESCRIPTION
**Pull request by sayaleeraut**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: api/restful/osd.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from api import Api
+
+
+class Osd(Api):
+
```
Comment (vamahaja) [inline api/restful/osd.py:5]: Add class description

File: tests/mgr/test_restful_operations_osd_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    if config.get("args"):
+        endpoint += f'/{config.get("args")}'
+    # Check if the endpoint needs to have "command"
+    if config.get("command"):
+        endpoint += "/command"
+    # Use GET method on the /osd or or /osd/<arg> or /osd/<arg>/command
+    # endpoint of the RESTful module
+    _code, _ = Osd(url=url, api=endpoint).get(username=username, key=key, check_sc=True)
+    if _code == status_code:
+        LOG.info("RESTful API validation is successful")
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_osd_endpoint.py:32]: Not required, please remove from all the places.

File: tests/mgr/test_restful_operations_osd_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    )
+    if _code == status_code:
+        LOG.info("RESTful API validation is successful")
+        return 0
+    raise OperationFailedError(
+        f"RESTful PATCH method for {endpoint} failed with status code:{_code}"
+    )
+
+
+def run(ceph_cluster, **kw):
```
Comment (vamahaja) [inline tests/mgr/test_restful_operations_osd_endpoint.py:79]: Add test case description

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: api/restful/osd.py (hunk 1 above)
Comment (sayaleeraut) [reply api/restful/osd.py:6]: Done

File: tests/mgr/test_restful_operations_osd_endpoint.py (hunk 2 above)
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_osd_endpoint.py]: Done

File: tests/mgr/test_restful_operations_osd_endpoint.py (hunk 3 above)
Comment (sayaleeraut) [reply tests/mgr/test_restful_operations_osd_endpoint.py:72]: Done

**Review by vamahaja — COMMENTED** [review]
Why separate suite? Can't it be part of the existing suite?

**Comment by vamahaja** [comment]
Kindly share tesrun logs

**Comment by sayaleeraut** [comment]
> Kindly share tesrun logs

Yes,

```
ceph-mgr RESTful module GET me   Use the GET method view osd information                        0:00:01.637215                   Pass                             
ceph-mgr RESTful module GET me   Use the GET method to view information about a osd id          0:00:01.725342                   Pass                             
ceph-mgr RESTful module GET me   Use the GET method to view implemented commands for the OSD    0:00:01.763652                   Pass                             
ceph-mgr RESTful module PATCH    Use the PATCH method to modify the config of OSD id            0:00:01.546300                   Pass                             
ceph-mgr RESTful module POST m   Use the POST method to run implemented command for the OSD i   0:00:01.645701                   Pass
```

Link - https://ocs3xstage-jenkins-csb-rhgsocs3x.apps.ocp-c1.prod.psi.redhat.com/view/Executor/job/qe-squid-test-executor-openstack/23/console 

Note: Will raise a separate PR for failures observed  with /pool tests

**Review by vamahaja — APPROVED** [review]

**Pull request merged**